Examples/Modules/ParticleBoundaryScrape/PICMI_inputs_wrapper_scrape.py

```diff
@@ -0,0 +1,152 @@
+# numerics parameters
+##########################
+
+# --- Number of time steps
+max_steps = 5
+diagnostic_intervals = "::5"
+
+# --- Grid
+nx = 128
+ny = 16
+
+xmin = 0.0
+ymin = 0.0
+xmax = D_CA
+ymax = D_CA / nx * ny
+
+number_per_cell_each_dim = [32, 16]
+
+##########################
+# physics components
+##########################
+
+v_rms_elec = np.sqrt(constants.kb * T_ELEC / constants.m_e)
+v_rms_ion = np.sqrt(constants.kb * T_INERT / M_ION)
+
+uniform_plasma_elec = picmi.UniformDistribution(
+    density = PLASMA_DENSITY,
+    upper_bound = [None] * 3,
+    rms_velocity = [v_rms_elec] * 3,
+    directed_velocity = [0.] * 3
+)
+
+uniform_plasma_ion = picmi.UniformDistribution(
+    density = PLASMA_DENSITY,
+    upper_bound = [None] * 3,
+    rms_velocity = [v_rms_ion] * 3,
+    directed_velocity = [0.] * 3
+)
+
+electrons = picmi.Species(
+    particle_type='electron', name='electrons',
+    initial_distribution=uniform_plasma_elec,
+    warpx_scrape_lo_x=1, warpx_scrape_hi_x=1
+)
+ions = picmi.Species(
+    particle_type='He', name='he_ions',
+    charge='q_e',
+    initial_distribution=uniform_plasma_ion,
+    warpx_scrape_lo_x=1, warpx_scrape_hi_x=1
+)
+
+##########################
+# numerics components
+##########################
+
+grid = picmi.Cartesian2DGrid(
+    number_of_cells = [nx, ny],
+    lower_bound = [xmin, ymin],
+    upper_bound = [xmax, ymax],
+    bc_xmin = 'dirichlet',
+    bc_xmax = 'dirichlet',
+    bc_ymin = 'periodic',
+    bc_ymax = 'periodic',
+    warpx_potential_hi_x=5,
+    lower_boundary_conditions_particles=['absorbing', 'periodic'],
+    upper_boundary_conditions_particles=['absorbing', 'periodic'],
+    warpx_max_grid_size = nx//4
+)
+
+solver = picmi.ElectrostaticSolver(
+    grid=grid, method='Multigrid', required_precision=1e-6
+)
+
+##########################
+# diagnostics
+##########################
+
+field_diag = picmi.FieldDiagnostic(
+    name = 'diag1',
+    grid = grid,
+    period = diagnostic_intervals,
+    data_list = ['rho_electrons', 'rho_he_ions', 'phi'],
+    write_dir = '.',
+    warpx_file_prefix = 'Python_particle_scraper_plt'
+)
+
+##########################
+# simulation setup
+##########################
+
+sim = picmi.Simulation(
+    solver = solver,
+    time_step_size = DT,
+    max_steps = max_steps,
+    verbose=True
+)
+
+sim.add_species(
+    electrons,
+    layout = picmi.GriddedLayout(
+        n_macroparticle_per_cell=number_per_cell_each_dim, grid=grid
+    )
+)
+sim.add_species(
+    ions,
+    layout = picmi.GriddedLayout(
+        n_macroparticle_per_cell=number_per_cell_each_dim, grid=grid
+    )
+)
+
+sim.add_diagnostic(field_diag)
+
+sim.step(max_steps - 1)
+
+# check scraped particle buffer part way through the simulation
+from pywarpx import _libwarpx
+
+n = _libwarpx.get_num_particles_impacted_boundary("electrons", 'x_hi')
+print("Number of electrons scraped:", n)
+weights = _libwarpx.get_particle_boundary_buffer("electrons", 'x_hi', 'w', 0)
```

Comment:
Would it be possible / desirable to set this up in such a way that it computes the total charge absorbed at each time step, and compares it to what it should be based on the simulation setup?

Reply:
I guess the best thing to do would be to make this test match yours exactly and then just also run the commands to check that the _libwarpx wrappers function properly. That way we can even use the same analysis file.